### PR TITLE
fix(fl-gradient-inversion): loss_grad logged before it is (+3 more)

### PR DESCRIPTION
**Pull request by andrewwhitecdw**
Small fixes in `FLGradientInversion/fl_gradient_inversion.py`:

#### fix: loss_grad logged before it is defined when grad_l2 is zero

**Fix:** Replace:
```
                loss = loss_aux
                if cfg["grad_l2"] > 0:
                    new_grad = self.sim_local_updates(
                        cfg=cfg,
                        trainer=local_trainer,
                        network=network,
                        inputs=inputs,
                        targets=targets_in,
                        use_sigmoid=True,
                        use_softmax=False,
                    )
                    loss_grad = 0
                    for a, b in zip(new_grad, self.grad_lst):
                        loss_grad += cfg["grad_l2"] * (torch.norm(a - b[1]))
                    loss = loss + loss_grad
```

with:
```
                loss = loss_aux
                loss_grad = torch.tensor(0.0, device=device)
                if cfg["grad_l2"] > 0:
                    new_grad = self.sim_local_updates(
                        cfg=cfg,
                        trainer=local_trainer,
                        network=network,
                        inputs=inputs,
                        targets=targets_in,
                        use_sigmoid=True,
                        use_softmax=False,
                    )
                    for a, b in zip(new_grad, self.grad_lst):
                        loss_grad += cfg["grad_l2"] * (torch.norm(a - b[1]))
                    loss = loss + loss_grad
```

#### fix: global model weights mutated across inversion iterations

**Fix:** Replace:
```
        params_after = trainer.network.state_dict()
        new_grad = []
        for name, _ in network.named_parameters():
            new_grad.append(params_after[name] - params_before[name])
        return new_grad
```

with:
```
        params_after = trainer.network.state_dict()
        new_grad = []
        for name, _ in network.named_parameters():
            new_grad.append(params_after[name] - params_before[name])
        network.load_state_dict(params_before)
        return new_grad
```

#### fix: DataLoader wrapped in extra list makes batch size ineffective

**Fix:** Replace:
```
        loader = DataLoader([data], batch_size=cfg["local_bs"])
```

with:
```
        loader = DataLoader(data, batch_size=cfg["local_bs"])
```

#### fix: DataLoader wrapped in extra list in sim_local_updates

**Fix:** Replace:
```
        trainer.data_loader = DataLoader([data], batch_size=cfg["local_bs"])
```

with:
```
        trainer.data_loader = DataLoader(data, batch_size=cfg["local_bs"])
```

### Files changed
- `FLGradientInversion/fl_gradient_inversion.py`

### CODE REVIEW

**Comment by andrewwhitecdw** [comment]
Closing this sweep-generated PR: sole commit is missing a valid Signed-off-by trailer. It does not meet the sweep requirements (single signed-off commit).
